docs/testplan/PFC_Snappi_Additional_Testcases.md

```diff
@@ -0,0 +1,65 @@
+|           09 | CONGESTION (PC + MACSEC) | **CONGESTION DETECTION ON PORT FAILURE ON PORTCHANNEL WITH MACSEC ENABLED:**<br>Test case to verify that DUT detects congestion on egress port-channel link on port-failure with MACSEC enabled on port-channel.                                                                                                          | - Minimum of Two x 100Gbps ingress and egress port with MACSEC and PFC-WD enabled.<br>- Min port set to 1 for this test <br>- Repeat with minimum 2x400Gbps ingress and egress ports as port-channel members. <br>- 1024B packet-size. | - 90% - Lossless Priority 3 and 4<br>- 10% - Lossy Background Flow[Prio0, 1, and 2]<br>- Shutdown one of the egress ports.<br>- Shutting one of egress port causing congestion on other egress port.<br>- Bring back the egress port online again <br> Repeat in loop. | - Single LC - Multiple ASIC<br>Multiple LC                                       | 1. Line-rate should be 200Gbps without any PFCs or packet drops. <br>2. When one of egress ports is shut,  PFCs should be sent to the ingress to slow down the incoming traffic.<br>3. The other egress port should receive all the Lossless Priority 3 and 4 traffic.<br>4. Miniscule drop for both lossless and lossy traffic while traffic is shifting to other egress port. <br> 5. When egress port is back-online, the traffic on egress ports should switch back with no loss or drops. |
+|           10 | PFCWD                    | **PFCWD ENABLED DROP MODE TEST:**<br> Purpose of the test case is to ensure that lossless priorities are dropped when PFCWD is enabled and egress ports faces PFC storm. DUT does not send PFCs out of ingress port and no drops for lossy traffic.                                                                       | - Two 100Gbps Ingress and single 100Gbps egress <br>- Repeat test with 2x400Gbps ingress and 1x400Gbps egress <br>-Repeat with IPv6. <br>- 1024B packet-size                                                                           | - 30% - Lossless Priority 3 and 4 per ingress LC <br>- 10% Lossy Background Flow [Priority 0, 1, and 2] per ingress LC. <br>- Enable PFCWD with "action" set to DROP(default)<br>- Send continuous PFC for lossless Priorities for configured interval.                | - Single LC - Multiple ASIC <br>- Single LC - Single ASIC <br>- Multiple LC      | 1. No lossy background flows dropped on DUT when the PFC is sent to DUT egress. Egress port is in 'stormed' mode. <br>2. Lossless flows are dropped on DUT. <br>3. PFCWD drop counter should increment. <br> 4. No PFCs are sent from DUT ingress to traffic transmitter. <br>5. After the PFC packets are stopped, the egress port is moves to "operational state" and storm-restored counter increments.                                                                                     |           

```

Comment:
before PFC-WD kicks, XOFF frames will be sent from ingress to the traffic source